iceoryx_hoofs/include/iceoryx_hoofs/internal/concurrent/fifo.inl

```diff
@@ -87,7 +87,7 @@ inline cxx::optional<ValueType> FiFo<ValueType, Capacity>::pop() noexcept
         // m_read_pos must be increased after reading the pop'ed value otherwise
         // it is possible that the pop'ed value is overwritten by push while it is read.
         // Implementing a single consumer fifo here allows us to use store.
-        m_read_pos.store(currentReadPos + 1, std::memory_order_relaxed);
+        m_read_pos.store(currentReadPos + 1, std::memory_order_release);
```

Comment:
OK, needed to write to `m_data` happens before this store and is synchronized with any load in a concurrent `pop`.

Resolve if this is the intention.